src/subscript/ofmvol2csv/ofmvol2csv.py

```diff
@@ -283,7 +283,10 @@ def parse_ofmtable(
             on_bad_lines="skip",  # pylint: disable=unexpected-keyword-arg
         )
 
-    data["DATE"] = pd.to_datetime(data["DATE"], dayfirst=True)
+    if version.parse(pd.__version__) >= version.parse("2.0"):
+        data["DATE"] = pd.to_datetime(data["DATE"], format="mixed", dayfirst=True)
+    else:
+        data["DATE"] = pd.to_datetime(data["DATE"], dayfirst=True)
```

Comment:
Can you confirm if the test that requires this branching is actually relevant?

Reply:
Unfortunately yes. pandas 1.5.3 doesn't accept format="mixed"  and will fail hard.

Reply:
@berland or do you mean we can prune some of the tests that failed without this branching? @rnyb is reaching out to OFM responsible to check if OFM has a specific date format.

Reply:
That is what I am asking, check whether the test code is doing meaningless tests. Parsing "2022-12-24" with dayfirst=True trying to get Pandas to think that this is day 12 and month 24 would be an example of such a meaningless test.

Reply:
Removing this branching will result in 3 failures. 

Assuming that OFM will not generate a mixed date format within a file, two of the tests can probably be removed:
https://github.com/equinor/subscript/blob/87c74b52819d90057bd9690c7f2db75152840759/tests/test_ofmvol2csv.py#L238-L251
https://github.com/equinor/subscript/blob/87c74b52819d90057bd9690c7f2db75152840759/tests/test_ofmvol2csv.py#L326-L340

But this branching also solve the issue that I mentioned in #525 

https://github.com/equinor/subscript/blob/87c74b52819d90057bd9690c7f2db75152840759/tests/test_csv2ofmvol.py#L224-L240

Reply:
Agree that the two first tests are probably not relevant, and is not a good enough reason to have the pandas-version-branch in the code.

Reply:
I just make a few changes:
- Revert pandas version branching
- Remove first test above
- Modify second test above to consistent date format since the test has other purpose (Well encoded as a column)
- Update the test so that it also use dayFirst=True, which make the 3rd test comparable.